woocommerce/woocommerce-order-source-attribution
# Add/wccom consent tracking

### Changes proposed in this Pull Request:

We're planning to [run our MVP](https://wp.me/pe2C5g-ur#comment-497) on [woocommerce.com](http://woocommerce.com/). In this PR, we hook into [woocommerce.com](http://woocommerce.com/) cookie consent management to ensure we're not tracking users who haven't consented.

This PR makes the following changes:

* Provide WCCOMTracking integration class. 
* Ensure woocommerce-order-attribute-source-js runs after sourcebuster by adding sourcebuster as a dependency script. wccom optimizes script loading. This ensures things don't break.
* Implement wccom-integration.js to trigger woocommerce_order_source_attribution.setAllowTrackingConsent if the wccom.canTrackUser returns true. This a fallback to server-side check; see https://a8c.slack.com/archives/C021P7KR98T/p1676897503785709?thread_ts=1676895080.006979&cid=C021P7KR98T
* Add woocommerce_order_source_attribution.removeTrackingCookies() to clean cookies when/if consent is revoked.

Closes #30.


### Checks:



### Screen recording:



https://user-images.githubusercontent.com/4209011/221167939-be8d2db2-d622-4912-897c-de9309bc4d22.mov




### Detailed test instructions:

1. First, make sure you're WCCOM locally - https://github.com/Automattic/woocommerce.com/blob/trunk/docs/setup.md
2. Clone this branch in the plugin folder of your WCCOM install and install the plugin.
3. Use this filter to force the cookie consent to open:
```
add_filter( 'body_class', function ( $classes ) {
	$classes[] = 'gdpr';

	return $classes;
} );
```
4. Open the woocommerce.test local site in incognito, accept cookies, and confirm that order attribution cookies are available.
5. Open the woocommerce.test the local site in incognito, disable analytics cookies, and confirm the order attribution cookies have been removed. 


### Changelog entry

> Add - handling wccom consent tracking
